src/transaction.rs

```diff
@@ -201,4 +244,43 @@ mod tests {
         let witness_script = format!("{}", descriptor.witness_script());
         assert_eq!(witness_script, "Script(OP_PUSHBYTES_33 0218845781f631c48f1c9709e23092067d06837f30aa0cd0544ac887fe91ddd166 OP_CHECKSIGVERIFY OP_PUSHBYTES_33 022222222222222222222222222222222222222222222222222222222222222222 OP_CHECKSIG)");
     }
+
+    #[test]
+    fn commitment_descriptor_to_witness_script() {
+        let X_0 = secp256k1::PublicKey::from_str(
+            "032a34617a9141231baa27bcadf622322eed1e16b6036fdf15f42a85f7250c4823",
+        )
+        .unwrap();
+        let R_0 = secp256k1::PublicKey::from_str(
+            "03ff65a7fedd9dc637bbaf3cbe4c5971de853e0c359195f19c57211fe0b96ab39e",
+        )
+        .unwrap()
+        .into();
+        let Y_0 = secp256k1::PublicKey::from_str(
+            "03b3ee07bb851fec17e6cdb2dc235523555dc3193c2ff6399ef28ce941bc57b2b4",
+        )
+        .unwrap()
+        .into();
+        let X_1 = secp256k1::PublicKey::from_str(
+            "03437a3813f17a264e2c8fc41fb0895634d34c7c9cb9147c553cc67ff37293b1cd",
+        )
+        .unwrap();
+        let R_1 = secp256k1::PublicKey::from_str(
+            "02b637ba109a2a844b27d31c9ffac41bfe080d2f0256eeb03839d66442c4ce0deb",
+        )
+        .unwrap()
+        .into();
+        let Y_1 = secp256k1::PublicKey::from_str(
+            "03851562dd136d68ff0911b4aa6b1ec95850144ddb939a1070159f0a4163d20895",
+        )
+        .unwrap()
+        .into();
+        let time_lock = 144;
+
+        let descriptor =
+            CommitTransaction::descriptor((X_0, R_0, Y_0), (X_1, R_1, Y_1), time_lock).unwrap();
+
+        let witness_script = format!("{}", descriptor.witness_script());
+        assert_eq!(witness_script, "Script(OP_SIZE OP_0NOTEQUAL OP_IF OP_PUSHBYTES_33 032a34617a9141231baa27bcadf622322eed1e16b6036fdf15f42a85f7250c4823 OP_CHECKSIG OP_SWAP OP_PUSHBYTES_33 03437a3813f17a264e2c8fc41fb0895634d34c7c9cb9147c553cc67ff37293b1cd OP_CHECKSIG OP_ADD OP_SWAP OP_DUP OP_IF OP_PUSHBYTES_2 9000 OP_CSV OP_VERIFY OP_ENDIF OP_ADD OP_PUSHNUM_3 OP_EQUALVERIFY OP_PUSHNUM_1 OP_ENDIF OP_TOALTSTACK OP_PUSHNUM_3 OP_PUSHBYTES_33 03437a3813f17a264e2c8fc41fb0895634d34c7c9cb9147c553cc67ff37293b1cd OP_PUSHBYTES_33 03b3ee07bb851fec17e6cdb2dc235523555dc3193c2ff6399ef28ce941bc57b2b4 OP_PUSHBYTES_33 03ff65a7fedd9dc637bbaf3cbe4c5971de853e0c359195f19c57211fe0b96ab39e OP_PUSHNUM_3 OP_CHECKMULTISIG OP_FROMALTSTACK OP_ADD OP_TOALTSTACK OP_PUSHNUM_3 OP_PUSHBYTES_33 032a34617a9141231baa27bcadf622322eed1e16b6036fdf15f42a85f7250c4823 OP_PUSHBYTES_33 03851562dd136d68ff0911b4aa6b1ec95850144ddb939a1070159f0a4163d20895 OP_PUSHBYTES_33 02b637ba109a2a844b27d31c9ffac41bfe080d2f0256eeb03839d66442c4ce0deb OP_PUSHNUM_3 OP_CHECKMULTISIG OP_FROMALTSTACK OP_ADD OP_PUSHNUM_1 OP_EQUAL)");
```

Comment:
Probably safe! 🤪

Reply:
I am totally trusting miniscript here. We should probably do a better test.

Reply:
We checked the new script together :)